src/neural/network_random.cc

```diff
@@ -50,16 +51,23 @@ class RandomNetworkComputation : public NetworkComputation {
     }
     inputs_.push_back(hash);
   }
+
   void ComputeBlocking() override {
     if (delay_ms_) {
       std::this_thread::sleep_for(std::chrono::milliseconds(delay_ms_));
     }
   }
 
   int GetBatchSize() const override { return inputs_.size(); }
+
   float GetQVal(int sample) const override {
     return (int(inputs_[sample] % 200000) - 100000) / 100000.0;
   }
+
+  float GetDVal(int sample) const override {
+    return (int(inputs_[sample] % 200000) - 100000) / 100000.0;
```

Comment:
This will always output the same D value as the Q value - which is wrong for a few reasons.
D should always be in the range 0 to 1.
Furthermore D should only ever be in the range 0 to (1-abs(Q)) since otherwise one of W or L must be negative.
Probably should HashCat in a constant, and then calculate its bounds correctly to know what divisor to use.